pkg/resource/layer_version/hooks.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package layer_version
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+// customPreDelete deletes all the previous versions of a
+//LayerVersion except the latest version
+
```

Comment:
Please remove this newline or Go documentation won't associate the comment with the method

Reply:
Done